src/Thinktecture.Relay.Server/Services/ConnectionStatisticsWriter.cs

```diff
@@ -30,15 +33,22 @@ await scope.ServiceProvider.GetRequiredService<IStatisticsService>()
 	}
 
 	/// <inheritdoc/>
-	public async Task UpdateLastActivityTimeAsync(string connectionId, CancellationToken cancellationToken = default)
+	public Task UpdateLastSeenTimeAsync(string connectionId, CancellationToken cancellationToken = default)
 	{
-		// TODO debounce writing to DB; keep connectionId and UtcNow in dictionary
-		// write to DB every {configurable duration} (default: 1 minute)
-		// additionally add this as a background service just like server statistics writer
+		_buffer[connectionId] = DateTimeOffset.UtcNow;
```

Comment:
This is not thread-safe. Better use `ConcurrentDictionary` instead.

Reply:
This seems not really resolved?

Reply:
This is using the `ConcurrentDictionary` now.